comfy/model_management.py

```diff
@@ -62,13 +64,18 @@ class CPUState(Enum):
 
 if args.cpu:
     cpu_state = CPUState.CPU
+    
+def is_cpu_with_ipex():
+    return cpu_state == CPUState.CPU and ipex_available and not mkl_is_crippled()
 
 def is_intel_xpu():
-    global cpu_state
-    global xpu_available
-    if cpu_state == CPUState.GPU:
-        if xpu_available:
-            return True
+    return cpu_state == CPUState.GPU and xpu_available
```

Comment:
I wonder if there should be an option to turn off the XPU explicitly here because potentially if someone wants to use the CPU for whatever reason, they can't if they are also using an Intel GPU.

Reply:
I've updated PR, so that now when user specifies `--cpu` flag `torch.xpu.is_available()` will never be run, so it won't use XPU code anywhere.

Overall, from design point ComfyUI overuses boolean flags. Better decision would be to use `--list-devices` (outputs list of available devices) with `--device=auto`, which selects the best device and fallbacks to CPU, if none found. Not only this relieves user from writing `--cpu` every time, this allows to specify exact GPU on multi-GPU/TPU/XPU/MPS/whatever systems.

Reply:
Looks good to me.